src/Donation/DonationRequestUtils.php

```diff
@@ -56,6 +59,10 @@ public function createFromRequest(Request $request, float $amount, int $duration
     {
         $clientIp = $request->getClientIp();
 
+        if ($donation = $this->session->get('donation_request')) {
```

Comment:
It can be moved up to the top as $clientIp is useless if it goes in the condition.